src/main/converter/opds.ts

```diff
@@ -176,15 +197,29 @@ export class OpdsFeedViewConverter {
             });
         }
 
+        if (feed.Metadata) {
+            page = {
+                numberOfItems: feed.Metadata.NumberOfItems,
+                itemsPerPage: feed.Metadata.ItemsPerPage,
+            };
+        }
+
         return {
             title,
             type,
             publications,
             navigation,
-            searchUrl: await this.getSearchUrlFromOpds1Feed(feed),
+            urls,
+            page,
         };
     }
 
+    private getUrlFromFeed(feed: OPDSFeed, linkRel: string): string {
+        return oc(feed.Links.find((link) =>
+            link.Rel && link.Rel.includes(linkRel),
+        )).Href(undefined);
```

Comment:
`oc` (`ts-optchain`) is useful for safely chaining "optional" (i.e. possibly `undefined` or `null`) object properties, in a type-strong manner.

However, in this particular case I don't think there is a strong justification for using `oc`, which instead seems to make the code harder to read, and to debug.

What about using "vanilla" coding style:
```
const linkWithRel = feed.Links.find((link) => link.Rel && link.Rel.includes(linkRel));
return linkWithRel ? linkWithRel.Href : undefined;
```